src/main/java/edu/cornell/kfs/concur/businessobjects/ValidationResult.java

```diff
@@ -59,8 +61,28 @@ public String getErrorMessagesAsOneFormattedString() {
                 result.append(KFSConstants.NEWLINE);
             }
         }
-        
-        return result.toString();
+
+        return addMessageHeaderAndTruncate(result.toString(), ConcurConstants.VALIDATION_RESULT_MESSAGE_MAX_LENGTH);
+    }
+
+    private String addMessageHeaderAndTruncate(String message, int maxLength) {
+        String errorMessagesString = addMessageHeader(message);
+        errorMessagesString = truncateMessageLength(errorMessagesString, maxLength);
```

Comment:
Good question. The requirements specs specifically asked for the error message to be truncated to 2000 characters. I will check with Sherry Wilson, the BA, to make sure the users understand this. Thank you!